### ISSUE TITLE
Unattended Installs

### ISSUE THREAD

**Issue by gavinfernandes2012** [chromebrew/chromebrew#1918]
## Unattended Installs
I've just noticed that chromebrew doesn't seem to have any built-in unattended install support, at least, it doesn't have it to the level that apt does. By this I mean that apt checks if it recognizes the package names before trying to install the first package, and that it automatically installs packages without confirmation in scripts or when the -y option is passed. The lack of the -y option can be worked around by `yes | crew install package1 package2 package3 package4`, but the other shortcomings are more complicated to work around manually and conveniently.

The early checking (in apt) warns the user if one of the packages cannot be found before proceeding to download and install other packages, hence the user can leave the computer to download and install the specified packages, but in crew as of now, if package3 from above is not a valid package name, the user will find out much later that neither package3 nor package4 has been installed, and if the user had left the computer for an extended period or put this in a script or sequence of "&&" separated commands, the next commands will not be executed.

## Versions and system information
chromebrew: `0.4.3`

ChromeOS: 
```
CHROMEOS_ARC_ANDROID_SDK_VERSION=25
CHROMEOS_ARC_VERSION=4625351
CHROMEOS_AUSERVER=https://tools.google.com/service/update2
CHROMEOS_BOARD_APPID={C166AF52-7EE9-4F08-AAA7-B4B895A9F336}
CHROMEOS_CANARY_APPID={90F229CE-83E2-4FAF-8479-E368A34938B1}
CHROMEOS_DEVSERVER=
CHROMEOS_RELEASE_APPID={C166AF52-7EE9-4F08-AAA7-B4B895A9F336}
CHROMEOS_RELEASE_BOARD=caroline-signed-mpkeys
CHROMEOS_RELEASE_BRANCH_NUMBER=46
CHROMEOS_RELEASE_BUILDER_PATH=caroline-release/R65-10323.46.0
CHROMEOS_RELEASE_BUILD_NUMBER=10323
CHROMEOS_RELEASE_BUILD_TYPE=Official Build
CHROMEOS_RELEASE_CHROME_MILESTONE=65
CHROMEOS_RELEASE_DESCRIPTION=10323.46.0 (Official Build) beta-channel caroline 
CHROMEOS_RELEASE_NAME=Chrome OS
CHROMEOS_RELEASE_PATCH_NUMBER=0
CHROMEOS_RELEASE_TRACK=beta-channel
CHROMEOS_RELEASE_VERSION=10323.46.0
DEVICETYPE=CHROMEBOOK
GOOGLE_RELEASE=10323.46.0
```

Model: CAROLINE (Samsung Chromebook Pro)

Architecture: `x86_64`

Basically it comes down to:
- [ ] Implement package name checks before any install attempts
- [ ] Add a -y option for "yes-to-all"
- [ ] Use -y by default when called from scripts

---

Now I can gladly help with this but it'll take some time cause I still haven't learned ruby, only say the word and I'll go ahead and try, but if you (the maintainers) want me to leave it to the pros I'm fine with that as well.

EDITS: Clarifications

**Comment by uberhacker** [comment]
@gavinfernandes2012: I'm not sure I follow.  Are you saying that if you have multiple arguments to crew install that all the arguments should be checked before it installs any of the packages?  Something like `yes | crew install ascii gawk rsync wim`?  Since the wim package doesn't exist, don't install anything?

**Comment by gavinfernandes2012** [comment]
Yeah that's exactly what I mean, (and it really hurts when you leave an install for half an hour only to come back and see it failed)

**Comment by satmandu** [comment]
One can shell script around this...
```
for i in pkg1 pkg2 pkg3 ... ; do [[ -e $i.rb ]] && (yes | crew install $i ) || echo "no $i package" ; done
```

**Comment by uberhacker** [comment]
It's even easier than that since crew accepts multiple parameters.
```
yes|crew install pkg1 pkg2 pkg3 ...
```
If you have a list of files named `packages.txt` such as:
```
pkg1
pkg2
pkg3
...
```
you can run `yes | crew install $(cat packages.txt | xargs)`.

**Comment by satmandu** [comment]
Yes but the question was what happens if one of the packages specified is not a package...

I think at the moment that causes crew to exit with a warning without installing any packages.

I suppose we could work around the issue in crew...

**Comment by uberhacker** [comment]
Garbage in and garbage out as they say. :)
